src/pages/patientView/clinicalInformation/ClinicalInformationPatientTable.tsx

```diff
@@ -33,14 +34,14 @@ export default class ClinicalInformationPatientTable extends React.Component<
         value: string;
     }): string {
         let ret: string;
-        switch (data.attribute) {
-            case 'Overall Survival (Months)':
-                ret = parseInt(data.value, 10).toFixed(0);
-                break;
-            default:
-                ret = data.value;
-                break;
+        const parsedFloat = parseFloat(data.value);
+
+        if (parsedFloat != parseInt(data.value)) {
+            ret = formatPercentValue(parsedFloat);
```

Comment:
i don't understand why we are using this formatPercentValue here.   also, why we are able to do away with special handling for Overall Survival (Months) case?

Reply:
Should we handle `Overall Survival (Months)` like any other value with months, like `Months from Diagnosis`? Do we strictly want month values to be integers without any decimal points? If not, maybe it's better to simply treat all floating point numbers the same way, like showing only 1 or 2 decimal points.

If we want more granularity, then we need to distinguish month attributes from other numerical attributes. Current production code only treats `Overall Survival (Months)` in a special way and probably incorrectly displays categorical attributes as `NaN`.

Reply:
@onursumer This was my thought process as well. The specific case of `Overall Survival (Months)` was made 6 years ago and I don't know how many attributes existed at that point. So @alisman I had thought it would be okay to handle `Overall Survival (Months)` like the other floating point attributes.

And Onur and I discussed using formatPercentValue here (https://github.com/cBioPortal/cbioportal-frontend/pull/4519#discussion_r1117223083)

Reply:
@alisman We discussed this last Friday during standup, and we seemed okay with things as long as they work and look good. I've corrected the NaN values from appearing and added a comment clarifying the use of `formatPercentValue`, so if things look okay to you this is hopefully good to go.